packages/modal/src/modal.ts

```diff
@@ -36,22 +35,51 @@ class Modal {
         const { showModal, hideModal, autoInitModal, isHidden } = initModal(modalElement, triggerButton, this.options);
 
         if (this.state === "open") {
-            showModal()
+            this.playOpenAnimation(); // Use animation when opening the modal
         }
-        autoInitModal()
-        this.showModal = showModal
-        this.hideModal = hideModal
-        this.isHidden = isHidden
+
+        this.showModal = this.playOpenAnimation.bind(this); // Update to use the open animation
+        this.hideModal = this.playCloseAnimation.bind(this); // Update to use the close animation
+        this.isHidden = isHidden;
+        
+        autoInitModal();
     }
 
+    // Predefined open animation
+    private playOpenAnimation(): void {
+        this.modalElement.style.display = "block"; // Ensure it's visible before animation
+        this.modalElement.animate([
```

Comment:
This is exactly what we need, but instead of a single animation like this, we need a list of animations, each with a name. In the modal options, we should be able to specify which animation we want to apply to the modal content when opening or closing it. (in options or by using a specific data attribute)